### PR TITLE
Nav: 1) New text manipulation commands, 2) new miscellaneous simple navigation commands, 3) move some existing commands into the CCR rule from the non-CCR rule.

### PR DESCRIPTION
**Pull request by alexboche**
Note_1: the new text manipulation commands use a slightly modified version of the standard dragonfly Function action which I called RemapArgsFunction and placed at the beginning of nav.py for now (should be eventually moved). This allows you to pass arguments in from the choice/extras that don't match the names of the arguments in the function signature. It allows you to pass in a remapping dictionary optionally to remap the names of the arguments. I have discussed this with Danesprite and I believe he will be adding this to the next release of dragonfly (possibly with some slight changes).

Note_2: one might say that a couple of the simple commands I added already exist but I think that something like shin ross wally deli (six syllables) is too long so I made shorter ones. I'm not wedded to any of these command names-- change the names however you like.

For some reason, this pull request is bundled with some old commits e.g. about Excel. I'm sorry I cannot figure out how to get rid of these. This pull requests should only involve changes to nav.py. I did "git pull upstream develop" to make sure I was up-to-date with the current develop branch and then I just changed one file nav.py so I don't understand why the Excel thing is getting involved with this. Please advise.

### CODE REVIEW

**Comment by LexiconCode** [comment]
Backup nav.py. Do a hard reset of the branch on the upstream. It will reset everything commit history current files exactly as the upstream branch. Then overwrite with your modified nav.py.

If you do git reset --hard <SOME-COMMIT> then Git will:

    Make your current branch back to point at <SOME-COMMIT>.
    Then make the files in your working tree and the index ("staging area") the same as the versions committed in <SOME-COMMIT>.

**Comment by alexboche** [comment]
I did git log to see what the latest command was from develop. As far as I can tell, it was commit number 262a2c673676b231f099c6f9fb2fd73de2ca3ac0 on Sunday, March 31 with author Mike Roberts with descriptions:  "Merge pull request #427 from mrob95/temp

    Create an action wrapper for read_selected_without_altering_clipboard."

I tried typing in git reset --hard 262a2c673676b231f099c6f9fb2fd73de2ca3ac0
I got the following message:
fatal: Could not parse object '262a2c673676b231f099c6f9fb2fd73de2ca3ac0'.
How should I proceed?

**Comment by LexiconCode** [comment]
Are you using a git client?

**Comment by alexboche** [comment]
I don't think it worked properly. Here is what I typed into power shell and the output I got. After I did this, I noticed that VisualStudio.py had been moved into a new location called : castervoice26/apps27/visualstudio.py as indicated in the output. I think this means I'm not current with the latest dictationtools/caster develop branch. "

PS C:\Users\alex\Desktop\caster-1> git fetch upstream
remote: Enumerating objects: 30, done.
remote: Counting objects: 100% (30/30), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 46 (delta 26), reused 27 (delta 26), pack-reused 16
Unpacking objects: 100% (46/46), done.
From https://github.com/dictation-toolbox/caster
   19d3bfe..262a2c6  develop    -> upstream/develop
 + bc5b2b8...c69094a master     -> upstream/master  (forced update)
PS C:\Users\alex\Desktop\caster-1> git reset --hard origin/develop
HEAD is now at cce2f07 Update and rename castervoice/apps/visualstudio.py to castervoice26/apps27/visualstudio.py
"

File: castervoice/lib/ccr/core/nav.py [hunk 1]
```diff
@@ -250,7 +347,43 @@ class Navigation(MergeRule):
...
+        "ecker":
+            R(Key("escape"), rspec= "ecker",  rdescript="Core: escape"),
+        "smack [<n>]":
+            R(Key("cs-left, del"), rspec="smack",  rdescript="Core: delete words to left") * Repeat(extra='n'),
+        "tack [<n>]":
+            R(Key("cs-right, del"), rspec="tack",  rdescript="Core: delete words to right") * Repeat(extra='n'),
+        "smack wally":
+            R(Key("s-left, del"), rspec="smack wally",  rdescript="Core: delete all words to left"),
+        "tack wally":
+            R(Key("s-right, del"), rspec="tack wally",  rdescript="Core: delete all words to right"),
```
Comment (mrob95) [inline castervoice/lib/ccr/core/nav.py]: These already exist as "escape" and "splat"

**Comment by mrob95** [comment]
Try running:
```
git checkout upstream/develop
git branch -D develop
git checkout -b develop
```

That will give you a clean develop branch

**Comment by alexboche** [comment]
Okay I tried that thanks. I made a new pull request that only has one commit. I tried to remove the duplicate commands and I moved the new supporting functions into navigation.py

**Pull request closed without merging**